### ISSUE TITLE
CLI does not include satellite assemblies for msbuild, roslyn, fsharp, nuget

### ISSUE THREAD

**Issue by nguerrera** [dotnet/cli#5007]
We need to figure out how to distribute satellite assemblies for msbuild and Roslyn with CLI in order to get csc and msbuild output translated in addition to messages coming directly from the CLI. I don't know about msbuild yet, but for Roslyn, it appears that to date these satellite assemblies are packaged only in the VS language pack, but not in nuget packages.

## Steps to reproduce

* `dotnet msbuild` on French OS installed with French VS language pack:

## Expected  behavior
Like desktop msbuild, output is translated:

```
D:>msbuild /nologo /v:m
MSBUILD : error MSB1011: Spécifiez le fichier projet ou solution à utiliser car ce dossier en contient plusieurs.
```

## Actual behavior

Output is not translated:
```
D:\Src\sdk>dotnet msbuild /nologo /v:m
MSBUILD : error MSB1011: Specify which project or solution file to use because this folder contains more than one project or solution file.
```

## Environment data
```
.NET Command Line Tools (1.0.0-preview4-004215)

Product Information:
 Version:            1.0.0-preview4-004215
 Commit SHA-1 hash:  7a0d3e255a

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\1.0.0-preview4-004215
```

**Comment by nguerrera** [comment]
@piotrpMSFT

**Comment by piotrpMSFT** [comment]
@jaredpar thoughts?

**Comment by jaredpar** [comment]
Not something we've really considered yet.  The way our sattelite assemblies are produced doesn't really mesh up with how we produce NuGet packages today.

How do other repos handle this? 

CC @OmarTawfik, @agocke

**Comment by nguerrera** [comment]
cc @cdmihai for msbuild perspective

**Comment by OmarTawfik** [comment]
@nguerrera @jaredpar I'm 99% sure there is a way to produce nuget packages from the build the produces satellite assemblies. Can you please tell me where do we produce the above mentioned nuget today? I'll try to look into it to confirm.

**Comment by cdmihai** [comment]
Same answer and @jaredpar. We currently include the satellite assemblies in the vsix package that goes into VS but not in the Nuget packages. We don't even produce localized builds for .net core, since our tooling that synchronizes the `xlf` files and generates the `resx` does not support .net core yet.

The Nuget documentation proposes two ways of providing localization files, either put satellites inside the `lib` directory, or use satellite packages: https://docs.nuget.org/ndocs/create-packages/creating-localized-packages
Should we all choose one variant so all repos do the same thing, or is it each repo for itself? :)
At first glance I'd go for satellite packages, as that minimizes the package size and duplicated data.
Either option require quite a bit of custom tooling.

Also, do satellite assemblies work in exactly the same way on .net core?
- Are the layout / naming conventions the same? Any .net Core ResourceManager gotchas?
- Do we need to recompile the satellite assemblies for .net core? In MSBuild's case, the full framework and .net core resource dlls could be the same. Since our build tooling does not support .net core, it would be awesome if we could just re-use the full framework satellite dlls. If not, can the `.resource` files be reused? What is the equivalent of `AL.exe` on .net core?

**Comment by nguerrera** [comment]
> Are the layout / naming conventions the same? 

The directory layout and naming conventions are the same. 


> Any .net Core ResourceManager gotchas?

No gotchas on the ResourceManager, but I just found one on Console vs non-UTF8 code page: https://github.com/dotnet/corefx/issues/10054. Expect dotnet/msbuild and dotnet/roslyn bugs on that.


> Do we need to recompile the satellite assemblies for .net core? 

I don't think so. I just tried copying VS's msbuild and roslyn satellites to dotnet's msbuild folder and my csc output is now localized even when running on .NET Core. (However, my msbuild output is still English and I don't yet know why. Expect a dotnet/msbuild bug on that.)

Looking in ildasm, a .resources.dll has a ref to mscorlib 4.0.0.0 for assembly level attributes, but .NET Core has a compatible mscorlib facade that forwards those attribute types. Other than that, the manifest resources have the same format. 


> The Nuget documentation proposes two ways of providing localization files, either put satellites inside the lib directory, or use satellite packages
> Should we all choose one variant so all repos do the same thing, or is it each repo for itself? :)

It would be nice if we all did the same thing, but at the same time, there may be a value judgment based on the size of your resource dlls. For example, the SDK repo currently has 22 strings. I'm not sure it's worth creating 10x as many nupkgs for that. I discussed with @srivatsn and we're leaning on just putting the satellites in lib. I'm not sure we need to impose a policy one way or the other. 

Do we think we absolutely must have roslyn and msbuild satellites in nupkgs or could we come up with another mechanism for chaining them in to the CLI installer that might be more expedient for VS 2017?


> What is the equivalent of AL.exe on .net core?

The SDK overrides the desktop AL.exe process and uses Roslyn to emit satellites: https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/GenerateSatelliteAssemblies.cs
